packages/teleport-uidl-resolver/src/utils.ts

```diff
@@ -440,7 +440,7 @@ const resolveDependency = (
 
     // ex: PrimaryButton component should be written in a file called primary-button
     // This is just a fallback for when the dependency path is not set by a project generator
-    const componentName = mappedElement.elementType
+    const componentName = mappedElement.semanticType ?? mappedElement.elementType
```

Comment:
are there situations where the mappedElement.semanticType exists? this is coming from the mapping files

Reply:
Yes in cases like custom mappings we need to consider this i feel, like a simple Button some external libraries like `chakra` in vue it seems to be `ca-button`. Not sure i am explaining correctly.